src/detectors/EEMC/Cluster_factory_EcalEndcapNClusters.h

```diff
@@ -15,6 +15,19 @@
 
 
 
+// Dummy factory for JFactoryGeneratorT
+class Association_factory_EcalEndcapNClustersAssociations : public JFactoryT<edm4eic::MCRecoClusterParticleAssociation> {
```

Comment:
For future reference:

1. We try to use a naming convention of <type_produced>_factory_<tag> so this class should probably be:
`MCRecoClusterParticleAssociation_factory_EcalEndcapNClusterAssociations`
(and yes, this is quite long!)

2. The `JFactoryGeneratorT` class actually takes a tag as the argument so you don't really need to define this class. You could do it in EEMC.cc as:

`app->Add( new JFactoryGeneratorT<JFactoryT<edm4eic::MCRecoClusterParticleAssociation>>("EcalEndcapNClustersAssociations"));`

I forgot to put the tag argument in the Mattermost message so sorry about that.

As a I say, this is for future reference. What you've done will work so I don't see a need to change it now.

Reply:
Thanks for mentioning the tag name in `JFactoryGeneratorT`. I think I can change it to make the code looks more clear.